### PR TITLE
KAYAK-1520 Upgrade deps: Confluent 7.3.1, Kafka 3.3.1

### PR DESCRIPTION
**Pull request by zcox**
[KAYAK-1520]

Required two changes:
1. Calling a non-deprecated `ConsumerRecord` constructor
2. Changing `sendOffsetsToTransaction` to accept a `ConsumerGroupMetadata` argument, matching a new method in the Java `KafkaProducer` API

That second change is an incompatible API change, so we'll probably want at least a minor version bump, if not a major.

[KAYAK-1520]: https://banno-jha.atlassian.net/browse/KAYAK-1520?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala [hunk 1]
```diff
@@ -68,13 +68,13 @@ case class ProducerOps[F[_], K, V](producer: ProducerApi[F, K, V]) {
...
       offsets: Map[TopicPartition, OffsetAndMetadata],
-      consumerGroupId: String,
+      groupMetadata: ConsumerGroupMetadata,
   )(implicit F: MonadError[F, Throwable]): F[Unit] =
     (for {
       _ <- producer.beginTransaction
       // should be no need to wait for RecordMetadatas or errors, since commitTransaction flushes and throws
       _ <- sendAndForgetBatch(records)
-      _ <- producer.sendOffsetsToTransaction(offsets, consumerGroupId)
+      _ <- producer.sendOffsetsToTransaction(offsets, groupMetadata)
```
Comment (Kazark) [inline core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala:77]: This is a change in the Java API, not a new overload?

**Review by Kazark — APPROVED** [review]

**Comment by amohrland** [comment]
Looks like the build's red due to a failed scalafmt check.

Shall we also bump the confluent docker images to 7.3.1?

**Review by amohrland — APPROVED** [review]

File: core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala (hunk 1 above)
Comment (zcox) [reply core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala:77]: They deprecated the old `sendOffsetsToTransaction` and added a new one. We can do the same thing if we really need to, or just make the small breaking change.

Comment (Kazark) [reply core/src/main/scala/com/banno/kafka/producer/ProducerOps.scala:77]: Ah, okay. Probably just easier to do what you did here.

**Pull request merged**